### ISSUE TITLE
[VarExporter][DoctrineBundle] Accessing unknown properties on lazy ghost objects generates notices instead exceptions

### ISSUE THREAD

**Issue by kiler129** [symfony/symfony#48970]
### Symfony version(s) affected

6.2.4

### Description

Initially this bug looked like an issue with EasyAdmin Bundle, as it is triggered by its code. The simplest way to do it is to add a field which doesn't exist on an entity, e.g.:

```
        yield TextField::new('type.thresholds_matches', 'Thresholds')
                       ->setVirtual(true)
                       ->setTemplatePath('admin/field/measurement_threshold.html.twig');
```

When `enable_lazy_ghost_objects` is disabled this works correctly and simply renders an empty field/custom template. However, the issue isn't actually in EAB but in `LazyGhostTrait` breaking the `PropertyAccessorInterface` promise:

https://github.com/symfony/symfony/blob/991c2bad292c4c4af17328c331507a5a7789512c/src/Symfony/Component/PropertyAccess/PropertyAccessorInterface.php#L69-L74

`->getValue()` should produce an exception when a given property doesn't exist, which EAB relies on: https://github.com/EasyCorp/EasyAdminBundle/blob/4db1860871809ca0be610ba5dab9be26ba165351/src/Field/Configurator/CommonPreConfigurator.php#L49-L53

### How to reproduce

PoC reproducer, pick your own entity of course. In my case the `type` field is configured as:

```php
    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false)]
    private ?MeasurementType $type = null;
```

The test code I used:
```php
<php declare(strict_types=1);

#[AsCommand(name: 'app:sandbox')]
class SandboxCommand extends Command
{
    public function __construct(private ManagerRegistry $doctrine, private PropertyAccessorInterface $propertyAccessor)
    {
        parent::__construct();
    }

    protected function execute(InputInterface $input, OutputInterface $output): int
    {
        $repo = $this->doctrine->getRepository(InanimateObjectMeasurement::class);
        $ent = $repo->findOneBy([]);

        dump($ent);
        dump($this->propertyAccessor->getValue($ent, 'value'));
        dump($this->propertyAccessor->getValue($ent, 'type.thresholds_matches'));
        
        return 0;
    }
```


With `enable_lazy_ghost_objects=false`:
```
 % bin/console app:sandbox -v
"183/88"

In PropertyAccessor.php line 453:
                                                                                                                               
  [Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException]                                                         
  Can't get a way to read the property "thresholds_matches" in class "Proxies\__CG__\App\Entity\Measurement\MeasurementType". 
```

With `enable_lazy_ghost_objects=true`:
```
 % bin/console app:sandbox -v
"183/88"

In LazyGhostTrait.php line 191:
                                                                                                                                                                                                                                       
  [ErrorException]                                                                                                                                                                                                                     
  User Notice: Undefined property: Proxies\__CG__\App\Entity\Measurement\MeasurementType::$thresholds_matches in /app/vendor/symfony/property-access/PropertyAccessor.php on line 428  
                                                                                                                                                                                                                                       
```

### Possible Solution

_No response_

### Additional Context

_No response_

**Comment by nicolas-grekas** [comment]
LazyGhostTrait cannot break the contract of PropertyAccessorInterface because it is not bound by the contract.
The issue is either in PropertyAccessor being too confident in having a __get meaning the property is reachable, or in easy admin relying on it while it shouldn't (why does a virtual field get fetched on a property?)

Anyway, can you please provide a repository with the working reproducer? That'd help a lot. Thanks.

**Comment by nickygerritsen** [comment]
We also run into this if you call `$propertyAccessor->isReadable` on a property that doesn't exist on a 'ghosted' Doctrine entity, you get this error. Not sure if it happens in general, but at least it happens here: https://github.com/DOMjudge/domjudge/blob/symfony-62/webapp/src/Controller/Jury/TeamController.php#L122 (note that on that branch lazy ghost objects are disabled since this would break our CI).

This is the backtrace for that error by the way:
```
ErrorException:
User Notice: Undefined property: Proxies\__CG__\App\Entity\Team::$num_contests in /domjudge/lib/vendor/symfony/property-access/PropertyAccessor.php on line 428

  at /domjudge/lib/vendor/symfony/var-exporter/LazyGhostTrait.php:193
  at Proxies\__CG__\App\Entity\Team->__get()
     (/domjudge/lib/vendor/symfony/property-access/PropertyAccessor.php:428)
  at Symfony\Component\PropertyAccess\PropertyAccessor->readProperty()
     (/domjudge/lib/vendor/symfony/property-access/PropertyAccessor.php:330)
  at Symfony\Component\PropertyAccess\PropertyAccessor->readPropertiesUntil()
     (/domjudge/lib/vendor/symfony/property-access/PropertyAccessor.php:227)
  at Symfony\Component\PropertyAccess\PropertyAccessor->isReadable()
     (src/Controller/Jury/TeamController.php:122)
  at App\Controller\Jury\TeamController->indexAction()
     (/domjudge/lib/vendor/symfony/http-kernel/HttpKernel.php:163)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw()
     (/domjudge/lib/vendor/symfony/http-kernel/HttpKernel.php:74)
  at Symfony\Component\HttpKernel\HttpKernel->handle()
     (/domjudge/lib/vendor/symfony/http-kernel/Kernel.php:184)
  at Symfony\Component\HttpKernel\Kernel->handle()
     (/domjudge/lib/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php:35)
  at Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()
     (/domjudge/lib/vendor/autoload_runtime.php:30)
  at require_once('/domjudge/lib/vendor/autoload_runtime.php')
     (public/index.php:5)  
```

**Comment by delboy1978uk** [comment]
We are upgrading a site to Symfony 6.3, we were using reflection in one of our classes to access the Connection class of an EntityManager, but now that fails because we do not have the real object and have this uninitialised lazy ghost instead. Any ideas?

**Comment by nicolas-grekas** [comment]
@kiler129 could you please wrap your reproducer into a repository to help me bootstrap this quickly?

**Comment by nickygerritsen** [comment]
I have a very minimal repo to reproduce this for my case. It comes down to:

```php
        $p    = PropertyAccess::createPropertyAccessor();
        $item = new \Proxies\__CG__\App\Entity\Item();
        $item->setName('test');
        dump($item);
        dump($p->isReadable($item, 'name')); // <-- Works
        dump($p->isReadable($item, 'missing')); // <-- Gives a notice
```

Repo can be found [here](https://github.com/nickygerritsen/lazy-ghost-bug).

Switching lazy ghost objects off makes this code work without a notice.

Let me know if you need anything else from me. My guess is this issue also resolves the original one.

**Comment by nicolas-grekas** [comment]
Should be fixed by #54194
